tests/test_parsing.js

```diff
@@ -1726,6 +1729,251 @@ describe('Parser', function() {
   });
 });
 
+describe('Parser modes', function() {
```

Comment:
There&nbsp;should&nbsp;probably be&nbsp;a&nbsp;test&nbsp;case for&nbsp;this:
```suggestion
describe('Parser modes', function() {
  it('should throw when invalid mode is specified, function() {
    expect(function() {
      parse('foo', {mode: 'invalid'})
    }).toThrow(TypeError);
  });
```

Reply:
Cool, added. I was thinking myself that we should really add coverage to be alerted to the need for tests like this, though I'm not sure what would be required to get nyc to recognize the PEG source.